hooks/useBalance.ts

```diff
@@ -23,12 +23,11 @@ export default function useBalance({
     isSuccess: isBalanceLoaded,
     isLoading: isFetchingBalance,
     refetch: refetchBalance,
-  } = useGetBalance({ // use cached global rpcClient inside
```

Comment:
as Lucas suggested, we don't use global for now.

Global cache's still available, users can decide whether use it or not.

Now, just pass rpcEndpoint to the clientResolver, pretty staight forward